lib/matplotlib/axes/_subplots.py

```diff
@@ -166,6 +166,23 @@ def _make_twin_axes(self, *args, **kwargs):
         self._twinned_axes.join(self, twin)
         return twin
 
+    def __repr__(self):
+        fields = []
+        if self.get_label():
+            fields += [f"label={self.get_label()!r}"]
+        titles = []
+        for k in ["left", "center", "right"]:
+            title = self.get_title(loc=k)
+            if title:
+                titles.append(f"{k!r}:{title!r}")
+        if titles:
+            fields += ["title={" + ",".join(titles) + "}"]
+        if self.get_xlabel():
+            fields += [f"xlabel={self.get_xlabel()!r}"]
+        if self.get_ylabel():
+            fields += [f"ylabel={self.get_ylabel()!r}"]
```

Comment:
Probably should update Axes3D, too? Or just do a `if hasattr(self, 'get_zlabel()')` or something?

Reply:
I think it would be better to have a custom repr on the Axes 3D than to inject knowledge of it here.  The subtle but pervasive interconnection is one of the things that makes maintaining Matplotlib hard.